### DISCUSSION TITLE
WGS 84 conversion to UE4 units and back

### DISCUSSION THREAD

**Discussion by Supaple-x** [microsoft/AirSim#3644] [Support Q&A] [answered]
## How to convert latitude, longitude, altitude (WGS 84) to UE4 Units and back?

I found out that units in AirSim are about 1 unit for 10cm.

**Comment by ahmed-elsaharti** [comment, marked as answer]
Hi @Supaple-x , AFAIK all the local frame locations reported by the AirSim APIs (Multirotorstate, etc) are in meters regardless of what units UE4 is using internally (see point 2 in [this reply](https://github.com/microsoft/AirSim/issues/565#issuecomment-339827160)).

If you're trying to convert your log's GPS points to xyz for the movement APIs you could use [this tool](https://geodesy.noaa.gov/NCAT/) linked in the article you mentioned. Moreover, if you use [PX4 (in SITL or HITL)](https://microsoft.github.io/AirSim/px4_setup/)+ QGroundControl in your simulation you could just create a mission in QGroundControl based on Lat+Lon waypoints and have the PX4 navigate those waypoints
